modules/gapi/samples/privacy_masking_camera.cpp

```diff
@@ -0,0 +1,215 @@
+        out_prims.clear();
+        const int BLOCK_SIZE = 24;
+        const auto cvt = [](cv::Rect rc) {
+            // Align the mosaic region to mosaic block size
+            const int dw = BLOCK_SIZE - (rc.width  % BLOCK_SIZE);
+            const int dh = BLOCK_SIZE - (rc.height % BLOCK_SIZE);
+            rc.width  += dw;
+            rc.height += dh;
+            rc.x      -= dw / 2;
+            rc.y      -= dh / 2;
+            return cv::gapi::wip::draw::Mosaic{rc, 24, 0};
+        };
+        for (auto &&rc : in_plate_rcs) { out_prims.emplace_back(cvt(rc)); }
+        for (auto &&rc : in_face_rcs)  { out_prims.emplace_back(cvt(rc)); }
+    }
+};
+
+} // namespace custom
+
+int main(int argc, char *argv[])
+{
+    cv::CommandLineParser cmd(argc, argv, keys);
+    cmd.about(about);
+    if (cmd.has("help")) {
+        cmd.printMessage();
+        return 0;
+    }
+    const std::string input = cmd.get<std::string>("input");
+    const bool no_show = cmd.get<bool>("noshow");
+    const bool run_trad = cmd.get<bool>("trad");
+
+    cv::GMat in;
+    cv::GMat blob_plates = cv::gapi::infer<custom::VehLicDetector>(in);
+    cv::GMat blob_faces  = cv::gapi::infer<custom::FaceDetector>(in);
+    cv::GArray<cv::Rect> rc_plates = custom::ParseSSD::on(blob_plates, in, 2);
+    cv::GArray<cv::Rect> rc_faces  = custom::ParseSSD::on(blob_faces, in, -1);
+    cv::GMat out = cv::gapi::wip::draw::render3ch(in, custom::ToMosaic::on(rc_plates, rc_faces));
+    cv::GComputation graph(in, out);
+
+    auto plate_net = cv::gapi::ie::Params<custom::VehLicDetector> {
+        cmd.get<std::string>("platm"),   // path to topology IR
+        cmd.get<std::string>("platw"),   // path to weights
+        cmd.get<std::string>("platd"),   // device specifier
+    };
+    auto face_net = cv::gapi::ie::Params<custom::FaceDetector> {
+        cmd.get<std::string>("facem"),   // path to topology IR
+        cmd.get<std::string>("facew"),   // path to weights
+        cmd.get<std::string>("faced"),   // device specifier
+    };
+    auto kernels = cv::gapi::kernels<custom::OCVParseSSD, custom::OCVToMosaic>();
+    auto networks = cv::gapi::networks(plate_net, face_net);
+
+    Avg avg;
+    cv::Mat out_frame;
+    std::size_t frames = 0u;
+    std::cout << "Reading " << input << std::endl;
+
+    if (run_trad) {
+        cv::Mat in_frame;
+        cv::VideoCapture cap(input);
+        cap >> in_frame;
+
+        auto exec = graph.compile(cv::descr_of(in_frame), cv::compile_args(kernels, networks));
+        avg.start();
+        do {
+            exec(in_frame, out_frame);
+            if (!no_show) {
+                cv::imshow("Out", out_frame);
+                cv::waitKey(1);
+            }
+            frames++;
+        } while (cap.read(in_frame));
+    } else {
+        auto pipeline = graph.compileStreaming(cv::compile_args(kernels, networks));
+        pipeline.setSource(cv::gapi::wip::make_src<cv::gapi::wip::GCaptureSource>(input));
```

Comment:
Can it work with `cv::VideoCapture`? I mean, if I have a source which should be preconfigured. In example Logitech C920. It works with 30FPS by default but for achieve 60FPS we need to set JPEG compressing, specific frame size (1280x720 if I'm not mistaken) and specific video backend (V4L2).

Reply:
`GCaptureSource` is built atop of `cv::VideoCapture`. I just did not want to expose `VideoCapture` itself to simplify the resource management (and make it less error-prone). May be what we might actually need is to support more `VideoCapture` parameters in the `GCaptureSource` constructor.

Note this API is still marked `wip` so